src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -153,6 +178,24 @@ private class MoreInfo
             public string? RemoteUrl;
         };
 
+        private void InterlockedSetFlags(Flags flags)
+        {
+            Debug.Assert(_syntax != null);
+
+            if (_syntax.IsSimple)
+            {
+                Debug.Assert(sizeof(Flags) == sizeof(ulong));
+                Interlocked.Or(ref Unsafe.As<Flags, ulong>(ref _flags), (ulong)flags);
+            }
+            else
+            {
+                lock (_info)
+                {
+                    _flags |= flags;
+                }
```

Comment:
For build-in (simple) parsers, it is safe to do an Interlocked update here, but a lock is still used for custom parsers in CreateHostString.

Reply:
Can you add that as a comment, including an explanation as to why?

Reply:
Done